src/wp-includes/class-wp-term-query.php

```diff
@@ -774,41 +775,47 @@ public function get_terms() {
 			return $this->terms;
 		}
 
-		$cache_key = $this->generate_cache_key( $args, $this->request );
-		$cache     = wp_cache_get( $cache_key, 'term-queries' );
+		if ( $args['cache_results'] ) {
+			$cache_key = $this->generate_cache_key( $args, $this->request );
+			$cache     = wp_cache_get( $cache_key, 'term-queries' );
+
+			if ( false !== $cache ) {
+				if ( 'ids' === $_fields ) {
+					$cache = array_map( 'intval', $cache );
+				} elseif ( 'count' !== $_fields ) {
+					if ( ( 'all_with_object_id' === $_fields && ! empty( $args['object_ids'] ) )
+						|| ( 'all' === $_fields && $args['pad_counts'] )
+					) {
+						$term_ids = wp_list_pluck( $cache, 'term_id' );
+					} else {
+						$term_ids = array_map( 'intval', $cache );
+					}
 
-		if ( false !== $cache ) {
-			if ( 'ids' === $_fields ) {
-				$cache = array_map( 'intval', $cache );
-			} elseif ( 'count' !== $_fields ) {
-				if ( ( 'all_with_object_id' === $_fields && ! empty( $args['object_ids'] ) )
-					|| ( 'all' === $_fields && $args['pad_counts'] )
-				) {
-					$term_ids = wp_list_pluck( $cache, 'term_id' );
-				} else {
-					$term_ids = array_map( 'intval', $cache );
-				}
+					_prime_term_caches( $term_ids, $args['update_term_meta_cache'] );
```

Comment:
See my note about that individual terms probably should be cached regardless of the `cache_results` setting to match behavior in WP_Query.